modules/matroska/video_stream_matroska.h

```diff
@@ -0,0 +1,135 @@
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "core/io/resource_loader.h"
+#include "core/templates/hash_map.h"
+#include "matroska.h"
+#include "scene/resources/image_texture.h"
+#include "scene/resources/video_stream.h"
+#include "scene/resources/video_stream_encoding.h"
+#include "servers/rendering/rendering_device.h"
+
+class VideoStreamMatroska : public VideoStream {
+	GDCLASS(VideoStreamMatroska, VideoStream);
+
+public:
+	void set_audio_track(int p_track) override;
+	Ref<VideoStreamPlayback> instantiate_playback() override;
+};
+
+class VideoStreamPlaybackMatroska : public VideoStreamPlayback {
+	GDCLASS(VideoStreamPlaybackMatroska, VideoStreamPlayback);
+
+private:
+	EbmlHeader header;
+	Segment segment;
+
+	String path;
+	const uint8_t *origin = nullptr;
+	const uint8_t *src = nullptr;
+
+	Ref<VideoStreamEncoding> video_stream_encoding = nullptr;
+
+	uint32_t width = 0;
+	uint32_t height = 0;
+
+	RenderingDevice *local_device;
+
+	RID video_session;
+
+	RID yuv_sampler;
+	RID yuv_shader;
+	RID yuv_pipeline;
+	Vector<RID> dst_yuv_pool;
+	Vector<RID> dst_rgba_pool;
+
+	size_t counter;
+	Ref<ImageTexture> image_texture;
+
+	bool playing = false;
+
+	void _skip_id(uint64_t p_id);
+
+	uint64_t read_id();
+	uint64_t read_size();
+
+	int64_t read_int();
+	uint64_t read_uint();
+	double read_float();
+	String read_string();
+
+	Error parse_ebml_header(EbmlHeader *r_header);
+	Error parse_segment(Segment *r_segment);
+
+	Error parse_seek_head(SeekHead *r_seak_head);
+	Error parse_segment_info(SegmentInfo *r_segment_info);
+	Error parse_tracks(Vector<Track> r_tracks);
```

Comment:
```suggestion
	Error parse_tracks(const Vector<Track> &p_tracks);
```
This shouldn't be prefixed with `r_` if it doesn't actually return anything